src/hb-subset-plan.cc

```diff
@@ -656,6 +736,11 @@ _fill_unicode_and_glyph_map(hb_subset_plan_t *plan,
     plan->codepoint_to_glyph->set (cp, gid);
     plan->unicode_to_new_gid_list.push (hb_pair (cp, gid));
   }
+
+  _unicode_closure (plan->unicode_to_new_gid_list,
+		    start,
+		    *plan->codepoint_to_glyph,
+		    unicode_to_gid);
```

Comment:
The "unicode_to_gid" function is only valid for codepoints in unicode_iterator on each particular invocation of _fill_unicode_and_glyph_map and can't be used as a general unicode to gid lookup. For example line 812 the function is fixed to just always return a specific gid because this function is invoked only for codepoints that map to that gid. We probably need to add an additional templated arg that provides a general mapper which is valid for all unicodes in the font.

Reply:
What an ouch... Yeah I think we might need that then, if we want to pursue this approach.

Reply:
Here's my pass at it: https://github.com/harfbuzz/harfbuzz/pull/4758

Reply:
Thanks. Rebased to use that.